src/pybind/mgr/ansible/module.py

```diff
@@ -673,6 +694,120 @@ def remove_host(self, host):
 
         return ARSChangeOperation(self.ar_client, self.log, operations)
 
+# JMO --------------------------------------------------------------------------
+    def add_stateless_service(self, service_type, spec):
+        """ Add a stateless service in the cluster
+
+        : service_type: Kind of service (nfs, rgw, mds)
+        : spec        : an Orchestrator.StatelessServiceSpec object
+
+        : returns     : Completion object
+        """
+
+        # Check service_type is supported
+        if service_type not in ["rgw"]:
+            raise orchestrator.OrchestratorError(
+                "{} service not supported".format(service_type))
+
+        # Add the hosts to the inventory in the right group
+        group = "{}s".format(service_type)
+        hosts = spec.service_spec.hosts
+        if not hosts:
+            raise orchestrator.OrchestratorError("No hosts provided."\
+                "At least one destination host is needed to install the RGW "\
+                "service")
+        self.ar_client.add_hosts_to_group(hosts, group)
+
+        # Limit playbook execution to certain hosts
+        limited = ",".join(hosts)
+
+        # Add the settings for this service
+        extravars = vars(spec.service_spec)
+
+        # Group hosts by resource (used in rm ops)
+        if service_type == "rgw":
+            resource_group = "rgw_zone_{}".format(spec.service_spec.rgw_zone)
```

Comment:
Same thing as with `group` name before, IMHO this should be defined in the RGWServiceSpec class/object.

Reply:
Very specific to Ansible... i can't see how to put this in a generic RGWSpec class.

Reply:
In that case, the Adapter could/should deal with it. While it ideally should be Service agnostic, any logic to couple specific service could also be placed there.

The logic itself wouldn't be much different from what is here, but as it's defined right now it's hard to reuse. And if you move the logic to a separate method, it will become a method handling with data from a single source (a `StatelessServiceSpec` instance from an Ansible perspective): hence it should be put inside a class.